tests/kernel/timer/timer_monotonic/src/main.c

```diff
@@ -69,19 +207,160 @@ void test_timer(void)
+	zassert_false(test_frequency_second(),
+			"test frequency failed");
+}
+
+/**
+ * @brief Test monotonic timer
+ *
+ * Validates monotonic timer's clock calibration for 1 ms.
+ *
+ * It reads the System clock’s h/w timer frequency value continuously
+ * using k_cycle_get_32() to verify its working and correctiveness.
+ * It also checks system tick frequency by checking the delta error
+ * between generated and system clock provided HW cycles per msec values.
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_cycle_get_32(), sys_clock_hw_cycles_per_sec()
+ */
+void test_timer_msec(void)
+{
+	u32_t t_last, t_now, i, errors;
+	s32_t diff;
+
+	errors = 0U;
+
+	TC_PRINT("k_ticks_to_cyc_floor32(1) = %d\n",
+			k_ticks_to_cyc_floor32(1));
+	TC_PRINT("sys_clock_hw_cycles_per_sec() = %d\n",
+			sys_clock_hw_cycles_per_sec());
+	TC_START("test monotonic timer");
+	t_last = k_cycle_get_32();
+
+	for (i = 0U; i < 1000000; i++) {
+
+		t_now = k_cycle_get_32();
+		if (t_now < t_last) {
+
+			diff = t_now - t_last;
+			TC_PRINT("diff = %d (t_last = %u : t_now = %u);"
+					"i = %u\n", diff, t_last, t_now, i);
+			errors++;
+		}
+
+	}
+	zassert_false(errors, "errors = %d\n", errors);
+	zassert_false(test_frequency_msecond(),
+			"test frequency failed");
+}
+
+/**
+ * @brief Test one shot timer
+ *
+ * Validates one shot timer's clock calibration for 1 ms.
+ *
+ * It reads the System clock’s h/w timer frequency value continuously
+ * using k_cycle_get_32() to verify its working and correctiveness.
+ * It also checks system tick frequency by checking the delta error
+ * between generated and system clock provided HW cycles per msec values.
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_cycle_get_32(), sys_clock_hw_cycles_per_sec()
+ */
+void test_oneshot_timer(void)
+{
+
+	u32_t t_last, t_now, i, errors;
+	s32_t diff;
+
+	errors = 0U;
+
+	TC_PRINT("k_ticks_to_cyc_floor32(1) = %d\n",
+			k_ticks_to_cyc_floor32(1));
+	TC_PRINT("sys_clock_hw_cycles_per_sec() = %d\n",
+			sys_clock_hw_cycles_per_sec());
+	TC_START("test monotonic timer");
+	t_last = k_cycle_get_32();
+
+	for (i = 0U; i < 1000000; i++) {
+
+		t_now = k_cycle_get_32();
+		if (t_now < t_last) {
+
+			diff = t_now - t_last;
+			TC_PRINT("diff = %d (t_last = %u : t_now = %u);"
+					"i = %u\n", diff, t_last, t_now, i);
+			errors++;
+		}
+	}
+	zassert_false(errors, "errors = %d\n", errors);
+	zassert_false(test_oneshottimer_frequency(),
+			"test one shot timer frequency failed");
+}
+
+
+/**
+ * @brief Test period timer
+ *
+ * Validates period timer's clock calibration for 1 ms.
+ *
+ * It reads the System clock’s h/w timer frequency value continuously
+ * using k_cycle_get_32() to verify its working and correctiveness.
+ * It also checks system tick frequency by checking the delta error
+ * between generated and system clock provided HW cycles per msec values.
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_cycle_get_32(), sys_clock_hw_cycles_per_sec()
+ */
+void test_period_timer(void)
```

Comment:
This is **also** doing exactly the same thing that test_timer() already did.  And like the oneshot one above, it seems to be claiming to be a k_timer test but it doesn't exercise any timers.